app/routers/chat.py

```diff
@@ -52,43 +57,106 @@ async def start_chat(user_input):
+            price_coin,
+            price_coin2,
+            price_coin3,
+        )
+
+    elif len(data_list) == 2:
+        messages = two_tokens_prompt(
+            docs,
+            extracted_currency_data,
+            currency_name,
+            currency_name2,
+            coin_symbol,
+            price_coin,
+            price_coin2,
+        )
+
+    else:
+        messages = [
+            {
+                "role": "system",
+                "content": f"""I am going to provide you Historical_currency_price_data in json format that contains information about a crypto currency and its historical data. Also a few Information_hex_pulse documents.\n
+                1) The user will ask about the historical price of any currency. You will have that currency information in the Historical_currency_price_data in json format. Use the timestamp in json data to tell the price of a currency of a specific date. \n
+                2) If the user asks any question or information that is present or related to the information in the provided documents then answer to that question using only these provided documents.\n
+                3) If the user asks some questions that are not related to these documents or you don't find in documents then respond to those question by using your knowledge.\n
+                4) Please respond with no salutations and don't refer to the provided documents while answering to user.\n
+                5) If the user asks about a date that is ahead of the today's date {formatted_date} or ask about the prediction of price like "will the hex price increase or decrease" then that use the future prices provided to you to predict the price or trend of a currency or price of currency from a future date. \n
+                Information_hex_pulse documents Starts:\n {docs}.\n Information_hex_pulse documents End\n
+                Historical_currency_price_data of {currency_name} :\n {extracted_currency_data}. \n Historical_currency_price_data of {currency_name} End.\n
+                provided future prices of {currency_name}:\n {future_data}\n
+                6) The current / today's price or price of currency_name: {currency_name} / currency_symbol: {coin_symbol} on {formatted_date} is {price_coin}. If the user asks about a currency price like "what is xrp price or price of bitcoin", tell the price from the current price of that currency.\n
+                7) If you have the price of hex and user asks about the price of hex then round the price of hex to 5 decimal places.
+                8) If you have the price of pulsechain (PLS) and user asks about the price of PLS then round the price of PLS to 7 decimal places.
+                """,
+            },
+        ]
+
     num_tokens_message = num_tokens_from_string(messages[0]["content"], "cl100k_base")
 
-    if num_tokens_message > 4040:
-        text_splitter = TokenTextSplitter(chunk_size=4040, chunk_overlap=0)
+    if num_tokens_message > 8000:
```

Comment:
this is correct value ?

Reply:
gpt-4 has limit of 8192 tokens so I set the limit to 8000